haystack/document_stores/deepsetcloud.py

```diff
@@ -46,6 +47,8 @@ def __init__(
                              If not specified, will be read from DEEPSET_CLOUD_API_ENDPOINT environment variable.
         :param similarity: The similarity function used to compare document vectors. 'dot_product' is the default since it is
                            more performant with DPR embeddings. 'cosine' is recommended if you are using a Sentence BERT model.
+        :param label_index: index for the evaluation set interface
```

Comment:
label_index is not being used. Please assign it to self.label_index

Reply:
It's passed into the `EvaluationSetClient` and used there.

Reply:
Ah ok, yep than it's being used. I stumbled upon `self.label_index = index` within this init. Please adjust this to `self.label_index = label_index`. Then we're good.